.github/workflows/python-ci.yml

```diff
@@ -0,0 +1,158 @@
+name: CI shared workflow
+
+on:
+  workflow_call:
+    inputs:
+      test-database:
+        required: false
+        type: boolean
+        default: false
+      database-name:
+        required: false
+        type: string
+        default: ""
+      database-user:
+        required: false
+        type: string
+        default: ""
+      database-password:
+        required: false
+        type: string
+        default: ""
+
+jobs:
+  StaticCodeAnalysis:
+    name: Static Code Analysis
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+            submodules: recursive
+            token: ${{ secrets.GH_TOKEN || github.token }}
+
+      - uses: pantos-io/ci-workflows/.github/actions/install-python-deps@v1
+
+      - name: StaticCodeAnalysis
+        run: |
+          make check
+
+  Format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+            submodules: recursive
+            token: ${{ secrets.GH_TOKEN || github.token }}
+
+      - uses: pantos-io/ci-workflows/.github/actions/install-python-deps@v1
+
+      - name: Formatting
+        id: formatter-check
+        run: make format-check
+
+  Lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+            submodules: recursive
+            token: ${{ secrets.GH_TOKEN || github.token }}
+
+      - uses: pantos-io/ci-workflows/.github/actions/install-python-deps@v1
+
+      - name: Lint
+        run: make lint
+
+  Sort:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+            submodules: recursive
+            token: ${{ secrets.GH_TOKEN || github.token }}
+
+      - uses: pantos-io/ci-workflows/.github/actions/install-python-deps@v1
+
+      - name: Sort
+        run: make sort-check
+
+  Bandit:
+    name: Bandit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+            submodules: recursive
+            token: ${{ secrets.GH_TOKEN || github.token }}
+
+      - uses: pantos-io/ci-workflows/.github/actions/install-python-deps@v1
+
+      - name: Bandit
+        run: make bandit
+
+  UnitTest:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: ${{ (inputs.test-database) && 'postgres:latest' || '' }}
+        env:
+          POSTGRES_DB: ${{ inputs.database-name }}
+          POSTGRES_USER: ${{ inputs.database-user }}
+          POSTGRES_PASSWORD: ${{ inputs.database-password }}
+          POSTGRES_HOST_AUTH_METHOD: trust
+        ports:
+          - 5432:5432
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+            submodules: recursive
+            token: ${{ secrets.GH_TOKEN || github.token }}
+
+      - uses: pantos-io/ci-workflows/.github/actions/install-python-deps@v1
+      
+      - name: UnitTest
+        run: |
+            make coverage
```

Comment:
Are not `make coverage` and `poetry run coverage xml` doing the same?

Reply:
Nope, `make coverage` is running the unit tests and generates the coverage report in a `.coverage` which is not understood by SonarCloud. `poetry run coverage xml` converts that file in a xml format which can then be uploaded to SonarCloud